### ISSUE TITLE
Using DINO3 as feature extractor

### ISSUE THREAD

**Issue by suvarnak** [ultralytics/ultralytics#22145]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

Ability to configure and use dino3 pretrained models as backbones. 
There seems to be a support for transformer based backbones in the code. https://github.com/ultralytics/ultralytics/blob/main/ultralytics/nn/modules/block.py
And as per [discussion](https://github.com/ultralytics/ultralytics/pull/18564
), "any" custom backbone can be used with the "TorchVision module that allows loading any torchvision model.

However, the  recent dinov3 backbones are available on torch hub 
https://github.com/facebookresearch/dinov3

### Use case

 DINO3 suite of models provide universal vision backbones (of different sizes) that can help achieve good performance across diverse domains, including web and satellite imagery.

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @suvarnak, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get started—an Ultralytics engineer will also assist you here shortly.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

For this Feature Request (DINOv3 backbones), it would be helpful if you could share:
- Target tasks you want to support (detect/segment/pose/classify) and expected feature map stages (e.g., strides/in_channels for P3–P5)
- Preferred integration path (via TorchVision module config vs. a dedicated backbone class) and example YAML usage
- Which pretrained variants you want first and expected input sizes
- Any minimal prototype showing backbone forward outputs plugged into a YOLO neck/head
- Basic benchmarks you aim to validate (throughput, params, mAP) and any constraints

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thanks again for your suggestion and for offering to submit a PR 🙌. Sharing the details above will help us guide the implementation and review efficiently.

**Comment by glenn-jocher** [comment]
Thanks for the suggestion—DINOv3 isn’t in TorchVision, so the existing `TorchVision` loader only covers torchvision models; see the `TorchVision` API and its args in the [module reference](https://docs.ultralytics.com/reference/nn/modules/block/#torchvision) and the YAML pattern in [Model YAML configuration (TorchVision integration)](https://docs.ultralytics.com/guides/model-yaml-config/#torchvision-integration). If you’d like to open a PR, a small `DINOv3` backbone wrapper that loads the hub model and returns feature maps (ideally multi-scale via a `split`-style option) plus a minimal YAML example, tests, and short docs would be great—please keep the change scoped and avoid edits to `default.yaml`. For inspiration on the expected interface, here’s the YAML pattern your wrapper should emulate:

```yaml
backbone:
  - [-1, 1, TorchVision, [768, convnext_tiny, DEFAULT, True, 2, False]]  # your DINOv3 wrapper would follow this shape contract
```

If you can share the target DINOv3 variants and intended output channel sizes/strides, we can help validate the design during review.

**Comment by Sompote** [comment]
please check my repo: Dino-Yolov12. https://github.com/Sompote/DINOV3-YOLOV12

**Comment by glenn-jocher** [comment]
Thanks for sharing—if you’d like this considered upstream, please open a PR (targeting YOLO11) that ports your DINOv3 backbone as a small wrapper with a minimal YAML and tests; the steps in [Custom Module Integration](https://docs.ultralytics.com/guides/model-yaml-config/#custom-module-integration) and the [Index module example for multi-scale features](https://docs.ultralytics.com/guides/model-yaml-config/#index-module-for-feature-selection) should help.

**Comment by dariussingh** [comment]
Hi @glenn-jocher

if this is still relevant to Ultralytics, can I work on this?

**Comment by glenn-jocher** [comment]
Yes, please go ahead, but since this issue is closed and marked stale, open a new feature request or a draft PR for visibility and target current `main` with a small, scoped `DINOv3` wrapper plus a minimal YAML example, tests, and docs, without adding new `default.yaml` arguments.

**Comment by MohitDAngrish** [comment]
is there any update on the dinov3 implementation?

**Comment by glenn-jocher** [comment]
No upstream update yet—there isn’t a `DINOv3` integration merged into current `main`, so if you’d like to move this forward, please open a new feature request or draft PR for visibility since this issue is closed and stale.
